Sources/BSON/Helpers.swift

```diff
@@ -110,7 +110,11 @@ internal protocol BSONMakeBytesProtocol: BSONBytesProtocol {
 
 extension Int : BSONBytesProtocol {
     internal func makeBytes() -> Bytes {
-        let integer = self.littleEndian
+        #if arch(s390x)
+            let integer = self.bigEndian
```

Comment:
Are you sure this (and the same for other makeBytes() implementations) is correct? It seems to me that the serialization code should always return littleEndian values (which is why there was `self.littleEndian` in the first place), because BSON itself should be serialized as little endian.

Reply:
@Obbut Thank you for looking at this.

I confirmed that this code serializes a value as little endian in a big-endian machine. As documented in https://developer.apple.com/documentation/swift/int/1538341-littleendian, x = x.bigEndian in a big-endian machine. If this code does not exist, the value is serialized as big endian.

For example, when an Int32 variable x is 5, 

Little-endian machine
x.littleEndian == 0x00000005 ==> [5, 0, 0, 0] (little-endian byte array)
x.bigEndian   == 0x05000000 ==> [0, 0, 0, 5] (big-endian byte array)

Big-endian machine
x.littleEndian == 0x05000000 ==> [0, 0, 0, 5] (big-endian byte array)
x.bigEndian   == 0x00000005 ==> [5, 0, 0, 0] (little-endian byte array)

So, we need to chose x.bigEndian in a big-endian machine.

Reply:
@Obbut How should we proceed?

Reply:
@tnakaike Is there a way I can test BSON on a big endian machine? As you may understand it is difficult for me to validate the workings of BSON on a big endian machine without access to one.

Reply:
@Obbut Yes. You can try a big-endian machine here https://developer.ibm.com/linuxone/.

Reply:
Thanks, I will do this today or tomorrow

Reply:
@Obbut I forgot one thing. In order to use swfit 4 on a mainframe, you need to build it by following this instruction https://github.com/linux-on-ibm-z/docs/wiki/Building-Swift?cm_mc_uid=48311614224114986550530&cm_mc_sid_50200000=1498483491. Sorry for your inconvenience.

Reply:
@tnakaike Is there an easier way to contact you? Are you on a public Slack / Discord perhaps?

Reply:
@Obbut Sorry for late response.

Let's use this slack https://developer.ibm.com/swift/. I will create a channel after you join if necessary.